glusterd2/commands/volumes/volume-option.go

```diff
@@ -113,13 +134,23 @@ func xlatorAction(c transaction.TxnCtx, txnOp txnOpType, volOp volumeOpType) err
 			reqOptions[key] = op.DefaultValue
 		}
 
-	}
-	var volinfo volume.Volinfo
-	if err := c.Get("volinfo", &volinfo); err != nil {
-		return err
+	case xlator.VolumeStart:
+		fallthrough
+
+	case xlator.VolumeStop:
+		for _, actor := range xlator.GetOptActors() {
+			if txnOp == txnDo {
+				fn = actor.Do
+			} else {
+				fn = actor.Undo
+			}
+			if err := fn(&volinfo, "", "", volOp, c.Logger()); err != nil {
+				return err
+			}
+		}
+		return nil
 	}
 
```

Comment:
Please add comment here that the code that follows applies only to `xlator.VolumeSet` and `xlator.VolumeReset` operations only.